embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbraceDeliveryService.kt

```diff
@@ -63,7 +63,7 @@ internal class EmbraceDeliveryService(
                         apiService.sendSession(
                             session,
                             onFinish
-                        )[SEND_SESSION_TIMEOUT, TimeUnit.SECONDS]
+                        )?.get(SEND_SESSION_TIMEOUT, TimeUnit.SECONDS)
```

Comment:
Is this correct? If `sendSession` fails to schedule a future then `onFinish` will result in data loss of the session

Reply:
It seems that onFinish is only executed on the finally blocked inside the executor's task. 
The only case where the future can be null is when the endpoint URL is null and that's not expected in any scenario. 
Anyway I think we already have a scenario of data loss here, because in EmbraceApiService we are calling onFinish() even when the request fails or is not executed because there is no network connection. 
We should review this path.

Reply:
This is an annoying leftover of the current API as we've validated that the URL isn't null. I should've taken care of this as part of the refactor and I'll do that later.